eng/common/scripts/update-docs-metadata.ps1

```diff
@@ -80,41 +80,53 @@ $pkgs = VerifyPackages -artifactLocation $ArtifactLocation `
   -releaseSha $ReleaseSHA `
   -continueOnError $True
 
-if ($pkgs) {
-  Write-Host "Given the visible artifacts, readmes will be copied for the following packages"
-  Write-Host ($pkgs | % { $_.PackageId })
+$targets = ($Configs | ConvertFrom-Json).targets
 
-  foreach ($packageInfo in $pkgs) {
-    # sync the doc repo
-    $semVer = [AzureEngSemanticVersion]::ParseVersionString($packageInfo.PackageVersion)
-    $rdSuffix = ""
-    if ($semVer.IsPreRelease) {
-      $rdSuffix = "-pre"
-    }
-
-    $readmeName = "$($packageInfo.PackageId.Replace('azure-','').Replace('Azure.', '').Replace('@azure/', '').ToLower())-readme$rdSuffix.md"
-    $readmeLocation = Join-Path $DocRepoLocation $DocRepoContentLocation $readmeName
-
-    if ($packageInfo.ReadmeContent) {
-      $adjustedContent = GetAdjustedReadmeContent -pkgInfo $packageInfo
-    }
-
-    if ($adjustedContent) {
-      try {
-        Push-Location $DocRepoLocation
-        Set-Content -Path $readmeLocation -Value $adjustedContent -Force
+foreach ($config in $targets) {
+  if ($config.mode -eq "Preview") { $includePreview = $true } else { $includePreview = $false }
+  $pkgsFiltered = $pkgs | ? { $_.IsPrerelease -eq $includePreview}
 
-        Write-Host "Updated readme for $readmeName."
-      } catch {
-        Write-Host $_
-      } finally {
-        Pop-Location
+  if ($pkgsFiltered) {
+    Write-Host "Given the visible artifacts, Readme updates against $($config.path_to_config) will be processed for the following packages."
+    Write-Host ($pkgsFiltered | % { $_.PackageId + " " + $_.PackageVersion })
+  
+    foreach ($packageInfo in $pkgsFiltered) {
+      # sync the doc repo
+      $semVer = [AzureEngSemanticVersion]::ParseVersionString($packageInfo.PackageVersion)
+      $rdSuffix = ""
+      if ($semVer.IsPreRelease) {
+        $rdSuffix = "-pre"
+      }
+  
+      $readmeName = "$($packageInfo.PackageId.Replace('azure-','').Replace('Azure.', '').Replace('@azure/', '').ToLower())-readme$rdSuffix.md"
```

Comment:
orthogonal but something to be aware of... for open telemetry some of the things now have @microsoft scope instead of @azure in the names (see https://github.com/Azure/azure-sdk-for-js/pull/12458/files). Eventually you might need to handle those as package names as well.

Reply:
Nah man. It's an easy fix. Will definitely get that in here too.

Reply:
I'm not happy with my  easy fix. Ugly as hell.

Reply:
Was the ugly fix part of this PR? I don't see any handling of this or at least didn't understand how it was handled in this PR.